src/java/org/apache/cassandra/db/guardrails/Guardrails.java

```diff
@@ -199,6 +200,19 @@ public final class Guardrails implements GuardrailsMBean
                    state -> CONFIG_PROVIDER.getOrCreate(state).getCompactTablesEnabled(),
                    "Creation of new COMPACT STORAGE tables");
 
+    /**
+     * Guardrail disabling the creation of a new table when default_time_to_live is set to 0 and
+     * compaction strategy is TimeWindowCompactionStrategy or its subclass. If this guardrail does not fail
+     * and such configuration combination is found, it will always warn.
+     */
```

Comment:
We could use JavaDoc links:
```suggestion
    /**
     * Guardrail disabling the creation of a new table when {@link TableParams.Option#DEFAULT_TIME_TO_LIVE} is set to 0
     * and compaction strategy is {@link TimeWindowCompactionStrategy} or its subclass. If this guardrail does not fail
     * and such configuration combination is found, it will always warn.
     */
```